controllers/budget_controller.py

```diff
@@ -1,76 +1,145 @@
 from sqlalchemy.orm import lazyload
 
 from models.Budget import Budget
+from models.Category import Category
 from models.ParentCategory import ParentCategory
 from session import session
+from prettytable import PrettyTable
 
-# GLOBAL VARIABLES
-global_user_id = None
-global_user_name = None
 
+def add_budget(user_id):
+    budget_name = input("Name of the new budget: ")
+    budget = Budget(name=budget_name, user_id=user_id)
+    session.add(budget)
+    session.commit()
+    # new_budget = list(session.query(Budget.id).order_by(Budget.id.desc()).first())[0]
+    new_budget = session.query(Budget.id).order_by(Budget.id.desc()).first()
+    return new_budget.id
 
-# DEFINE LOGIC HERE
-
-# Show user's budget
-def show_budget():
-    # refer to global variables inside function
-    global global_user_id
-    global global_user_name
-
-    # Try to retrieve the user's budget
-    budget_instance = session.query(Budget).filter_by(user_id=global_user_id).first()
-
-    # If user has no budgets
-    if budget_instance is None:
-        print("\nWhoops, you don't have any budgets yet. Shall we create one?")
-        add_budget()
 
-    # else == User does have at least 1 budget
+def select_budget(user_id):
+    budgets = session.query(Budget).filter(Budget.user_id == user_id).all()
```

Comment:
można to zapisać:

budgets = user.budgets

trzeba tylko wcześniej dać funkcji obiekt usera - dlatego myślę, że dobrze byłoby zacząć zwracać / dawać funkcjom obiekty, a nie tylko ich id

Reply:
Poprawione